### ISSUE TITLE
CollectionView SelectedItems Two Way Binding doesn't work

### ISSUE THREAD

**Issue by BlueRaja** [dotnet/maui#23358]
### Description

If you bind `SelectedItems` with `Mode=TwoWay`, selecting an item should add it to the list.  But it doesn't.

The [documentation](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/collectionview/selection?view=net-maui-8.0) says "The default binding is one-way", implying that two-way should be possible.

### Steps to Reproduce

 ```
  <CollectionView x:Name="collectionView" 
                  IsGrouped="True" 
                  SelectionMode="Multiple"
                  SelectedItems="{Binding SelectedItems, Mode=TwoWay}">
    ...
  </CollectionView>
```
```
public partial class MyContentView : ContentView
{
    public ObservableCollection<char> SelectedItems => new ObservableCollection<char>();

    public MyContentView()
    {
        InitializeComponent();
        BindingContext = this;
    }
}
```

### Version with bug

8.0.61 SR6.1

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

all

### Did you find any workaround?

No. You can workaround this in one direction, but not the other, because `collectionView.SelectedItems.Add()` doesn't actually work. I'll leave the code to fix the other direction here, for posterity:

Hook `SelectionChanged`, manually determine difference between `e.PreviousSelection` and `e.CurrentSelection`, then manually add/remove items from `SelectedItems`

```
foreach (var addedItem in e.CurrentSelection.Except(e.PreviousSelection).Cast<KanjiSelectionListNewModelKanji>())
{
    SelectedKanji.Add(addedItem.Character);
}
foreach (var removedItem in e.PreviousSelection.Except(e.CurrentSelection).Cast<KanjiSelectionListNewModelKanji>())
{
    SelectedKanji.Remove(removedItem.Character);
}
```

**Comment by BlueRaja** [comment]
I don't think this is the same as #8435 because changing it to `ObservableCollection<object>` does not work around the issue.

**Comment by davidce** [comment]
Change  "ObservableCollection<char>" to "IList<object>" works for me

**Comment by BlueRaja** [comment]
@davidce Nope. The `SelectedItems` binding still does absolutely nothing.

**Comment by BlueRaja** [comment]
ok with the above hint, I got this working:

* The collection needs to be `ObservableCollection<object>`
* It needs to be a property, not a field
* It needs to be on the ViewModel, not the ContentView
* With grouping enabled, the types of the items will be the enumerated-type of the group, even if you use an `ItemTemplate`. _(In retrospect this should've been obvious, but I guess I had a brain fart. Support for proper typing on the collection would've caught this...)_
* For some reason (???) the ObservableCollection seems to be cloning the selected items, meaning you can't use reference-equality when comparing them

So I guess this bug can be closed
